src/main/java/org/zalando/nakadi/service/subscription/StreamParameters.java

```diff
@@ -41,25 +43,31 @@
 
     private final Client consumingClient;
 
-    private StreamParameters(
-            final int batchLimitEvents, @Nullable final Long streamLimitEvents, final long batchTimeoutMillis,
-            @Nullable final Long streamTimeoutSeconds, @Nullable final Integer batchKeepAliveIterations,
-            final int maxUncommittedMessages, final long commitTimeoutMillis, final Client consumingClient)
-            throws UnprocessableEntityException {
-        if (batchLimitEvents > 0) {
-            this.batchLimitEvents = batchLimitEvents;
-        } else {
-            throw new UnprocessableEntityException("batch_limit can't be lower than 1");
+    private final List<EventTypePartition> partitions;
+
+    private StreamParameters(final UserStreamParameters userParameters, final long commitTimeoutMillis,
+                             final Client consumingClient) throws WrongStreamParametersException {
+
+        this.batchLimitEvents = userParameters.getBatchLimit().orElse(1);
+        if (batchLimitEvents <= 0) {
+            throw new WrongStreamParametersException("batch_limit can't be lower than 1");
         }
-        this.streamLimitEvents = Optional.ofNullable(streamLimitEvents).filter(v -> v != 0);
-        this.batchTimeoutMillis = batchTimeoutMillis;
+
+        this.streamLimitEvents = userParameters.getStreamLimit().filter(v -> v != 0);
+
```

Comment:
This is a very minor point, but I don't understand why there are empty lines here (except between lines 67-68, and 70-71)

Reply:
I just wanted to improve readability using empty lines :) but it seems that it brings confusion - so I will remove empty lines.

Reply:
https://github.com/zalando/nakadi/pull/848/commits/cafbbecadb02e7908b7eeecd47979c9bcd85d1f6